service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/CbsSessionHandler.java

```diff
@@ -131,7 +132,7 @@ public DeliveryState handleAuthenticationResponseMessage(int status, String desc
         }
         else
         {
-            IotHubException e = new IotHubException(status, description);
+            IotHubException e = IotHubExceptionManager.mapException(status, description);
```

Comment:
This was a small bug in my earlier work where I didn't actually map to the more specific IotHubException subtype. Now this will throw an IotHubUnauthorizedException for instance.